tests/test_type_inference/test_attribute.py

```diff
@@ -106,3 +107,137 @@ def test_attribute_self_bind():
     x = [ti.lookup_typevar(node, node.name) for node
          in module.nodes_of_class(astroid.AssignName)][0]
     assert str(ti.type_constraints.resolve(x).getValue()) == "typing.List[int]"
+
+
+def test_subscript_attribute():
+    program = \
+        '''
+        class A:
+            def __init__(self):
+                self.lst = []
+                self.lst[0] = 1
+                
+        a = A()
+        x = a.lst[0]
+        '''
+    module, ti = cs._parse_text(program, reset=True)
+    for assgn_node in module.nodes_of_class(astroid.AssignName):
+        if assgn_node.name == 'x':
+            x = ti.lookup_typevar(assgn_node, assgn_node.name)
+            assert ti.type_constraints.resolve(x).getValue() == int
+
+
+def test_obj_list_attribute():
+    program = \
+        '''
+        class A:
+            def __init__(self):
+                self.name = 'abc'
+
+        lst = [A(), A()]
+        x = lst[0].name
+        '''
+    module, ti = cs._parse_text(program, reset=True)
+    for assgn_node in module.nodes_of_class(astroid.AssignName):
+        if assgn_node.name == 'x':
+            x = ti.lookup_typevar(assgn_node, assgn_node.name)
+            assert ti.type_constraints.resolve(x).getValue() == str
+
+
+def test_unknown_class_attribute():
+    program = \
+        '''
+        def foo(a, x):
+            y = x.name
+            y = 3
+            z = x.name
+        '''
+    module, ti = cs._parse_text(program, reset=True)
+    for assgn_node in module.nodes_of_class(astroid.AssignName):
+        if assgn_node.name == 'z':
+            z = ti.lookup_typevar(assgn_node, assgn_node.name)
+            assert ti.type_constraints.resolve(z).getValue() == int
+
+
+def test_unknown_class_attribute2():
+    program = \
+        '''
+        def foo(x):
+            return x.name + 2
+            
+        def bar(y):
+            z = foo(y)
+        '''
+    module, ti = cs._parse_text(program, reset=True)
+    for binop_node in module.nodes_of_class(astroid.BinOp):
+        assert binop_node.inf_type.getValue() == int
+    for assgn_node in module.nodes_of_class(astroid.AssignName):
+        if assgn_node.name == 'z':
+            z = ti.lookup_typevar(assgn_node, assgn_node.name)
+            assert ti.type_constraints.resolve(z).getValue() == int
+
+
+def test_unknown_class_subscript_attribute():
+    raise SkipTest("Currently, the inferred type of the z variable is str")
```

Comment:
This is related to the issue described in #531 (because the first of all the `__getitem__` signatures apparently returns a `str`).